File: fizzbuzz/java/src/main/java/org/example/FizzBuzz.java
package org.example;

public class FizzBuzz {
    public static String execute(int number) {
        if(number == 0) {
            return "0";
        }
        boolean isDivisibleByThree = number % 3 == 0;
        boolean isDivisibleByFive = number % 5 == 0;

        if (isDivisibleByThree && isDivisibleByFive) {
            return "fizzbuzz";
        } else if (isDivisibleByThree) {
            return "fizz";
        } else if (isDivisibleByFive) {
            return "buzz";

        }

        return Integer.toString(number);
    }
}

File: fizzbuzz/typescript/src/core/fizzbuzz.ts
export const whoFizzBuzzElementIs = (num: number) => {
    const numberStringify = num.toString();
    const isDivisibleByThree = num % 3 === 0
    const isDivisibleByFive = num % 5 === 0
    let fizzbuzz = ""

    if (isDivisibleByThree) {
        fizzbuzz += "fizz"
    }

    if(numberStringify.includes("3")) {
        fizzbuzz += "fizz"
    }

    if (isDivisibleByFive) {
        fizzbuzz += "buzz"
    }

    if(numberStringify.includes("5")) {
        fizzbuzz += "buzz"
    }

    return fizzbuzz === "" ? num : fizzbuzz
}

export const fizzbuzz = () => {
    const arr = Array.from(Array(100).keys()).map(num => num + 1)
    const fizzBuzzResult: (string | number)[] = arr.map(num => whoFizzBuzzElementIs(num))

    return fizzBuzzResult
};